### DISCUSSION TITLE
Poll: Lightning vs Hyper-SD

### DISCUSSION THREAD

**Discussion by mashb1t** [lllyasviel/Fooocus#2813] [Polls]
Quick check for https://github.com/lllyasviel/Fooocus/pull/2812 / https://github.com/lllyasviel/Fooocus/pull/2812 comparing Lightning 4 steps LoRA with Hyper-SD 4 steps LoRA.
Images of each round were created using the same model / config / seed / styles / etc., only difference is the performance setting.

~I'll post which one is which in a few days.~
A is Lightning, B is Hyper-SD.

You can find the comparison between Extreme Speed (LCM) and Lightning here: https://github.com/lllyasviel/Fooocus/discussions/2416

<details><summary>Round 1</summary>
<p>

A
![image - 2024-04-26T224329 114](https://github.com/lllyasviel/Fooocus/assets/9307310/402855e9-cec7-4633-ad4c-665002708ef5)

B
![image - 2024-04-26T224341 127](https://github.com/lllyasviel/Fooocus/assets/9307310/dd2de9ea-3df8-405f-b8cf-e572e6bec627)

</p>
</details> 

<details><summary>Round 2</summary>
<p>

A
![image - 2024-04-26T230410 134](https://github.com/lllyasviel/Fooocus/assets/9307310/97048f09-c1e2-4982-88e9-981e11b88b21)

B
![image - 2024-04-26T230357 726](https://github.com/lllyasviel/Fooocus/assets/9307310/1c96a105-0bfc-4305-9ec3-54d5b55c1463)

</p>
</details> 

<details><summary>Round 3</summary>
<p>

A
![image - 2024-04-26T224656 108](https://github.com/lllyasviel/Fooocus/assets/9307310/4910151e-47e3-40f9-95fa-418ab2d63937)

B
![image - 2024-04-26T224638 994](https://github.com/lllyasviel/Fooocus/assets/9307310/8cd1f13b-87ff-4bc8-b920-07c7d1ba5631)

</p>
</details> 

<details><summary>Round 4</summary>
<p>

A
![image - 2024-04-26T224719 212](https://github.com/lllyasviel/Fooocus/assets/9307310/320192f1-dec5-446c-ad0d-9eb75cdcbda5)

B
![image - 2024-04-26T224735 350](https://github.com/lllyasviel/Fooocus/assets/9307310/52bbe391-4116-4b62-9d9b-2a3f90530ffd)

</p>
</details> 

<details><summary>Round 5</summary>
<p>

A
![2024-04-26_22-18-38_7311](https://github.com/lllyasviel/Fooocus/assets/9307310/2ceb54a1-8bf9-4665-ab13-0ed58967696d)

B
![2024-04-26_22-18-23_7311](https://github.com/lllyasviel/Fooocus/assets/9307310/2daca592-abca-4eb0-b923-d5e0f2bbf867)

</p>
</details>

**Comment by xhoxye** [comment]
`sampler_name = 'ddim',
scheduler_name = 'sgm_uniform'`
Judging by my own test results, this combination of parameters is not optimal.

**Comment by xhoxye** [comment]
Fooocus v2.3.1
juggernautXL_version6Rundiffusion.safetensors
styles = none
seed = 123
<details>
<summary>A female magician wearing a magic hat,Sitting at the street coffee table drinking coffee</summary>
 
![light](https://github.com/lllyasviel/Fooocus/assets/129571231/4faaa517-1bd6-4f57-a2cc-2458edd587f5)
![2024-04-27_08-39-42_7760](https://github.com/lllyasviel/Fooocus/assets/129571231/394b8ac4-eb59-43d2-9677-e30ca77251d1)

![euler_a-8](https://github.com/lllyasviel/Fooocus/assets/129571231/d2bc2dd3-aada-4262-90b5-44b790a32f6f)
![2024-04-27_08-38-37_7855](https://github.com/lllyasviel/Fooocus/assets/129571231/7db8e006-d46c-4a81-a94e-3b76c7b4045e)

![ddim-4](https://github.com/lllyasviel/Fooocus/assets/129571231/669a1674-457b-4927-b23d-bd6a2fda7006)
![2024-04-27_08-37-24_5784](https://github.com/lllyasviel/Fooocus/assets/129571231/b55edda3-9cf4-458d-9f20-a10d8d869014)

 
</details>

**Comment by wangfeng35** [comment]
It seems current version tends to "burn" the image when CFG is high. 
[According to the team](https://huggingface.co/ByteDance/Hyper-SD#news%F0%9F%94%A5%F0%9F%94%A5%F0%9F%94%A5), their CFG-Preserved Hyper-SD15/SDXL that facilitate negative prompts and larger guidance scales (e.g. 5~10) will be coming soon! Looking forward to it.

**Reply by wangfeng35** [reply]
They just uploaded [8-steps CFG-preserve LoRA for SDXL](https://huggingface.co/ByteDance/Hyper-SD/commit/ebed5c7e77403044ba523e81a428cf599f4a9463) (support 5~8 guidance scales).  4-steps version will be coming soon.

**Comment by Irmagon** [comment]
My preset, after ~200 generation:

"default_model": "juggernautXL_v8Rundiffusion.safetensors",
"default_loras": [["Hyper-SDXL-4steps-lora.safetensors",**0.8**]],
"default_cfg_scale": 1,
"default_sample_sharpness": 1.0,
"default_overwrite_step": 4,
"default_sampler": "**dpmpp_sde_gpu**",
"default_scheduler": "**karras**",
"default_performance": "Speed",
"default_styles": ["Fooocus V2"],
"default_aspect_ratio": "1024*1024",

![2024-04-27_15-14-51_6524](https://github.com/lllyasviel/Fooocus/assets/36661892/c0f7ae6d-530e-42b9-80f1-81c46b5c80e3)

**Comment by mashb1t** [comment]
@Irmagon thanks for the hint, i can confirm that these settings work way better contrary to the suggested ones.

LoRA weight 0.8
sampler dpmpp_sde_gpu
scheduler karras

all one shot images, no best-of-4, tested with Juggernaut v8 Rundiffusion:

![image - 2024-04-30T143837 994](https://github.com/lllyasviel/Fooocus/assets/9307310/8fca80c9-4010-40ad-afd3-e5b44edb1a91)
![image - 2024-04-30T144034 848](https://github.com/lllyasviel/Fooocus/assets/9307310/8ae191c8-97d3-4b3d-8f23-88401f2e3ad2)
![image - 2024-04-30T144221 282](https://github.com/lllyasviel/Fooocus/assets/9307310/898dc14d-6197-4e97-89db-f17adf96decb)
![image - 2024-04-30T144417 436](https://github.com/lllyasviel/Fooocus/assets/9307310/98f9db4f-e45c-4f36-9dcc-fc47bd8b7b3d)
![image - 2024-04-30T144821 082](https://github.com/lllyasviel/Fooocus/assets/9307310/639f4a3d-7756-48a5-8997-3a31bbd46f56)

**Comment by ILikeToasters** [comment]
Looks like they have some new 8-step LoRAs out with 4-step ones coming soon. 
<img width="391" alt="Screenshot 2024-04-30 131723" src="https://github.com/lllyasviel/Fooocus/assets/4357657/cd2c075d-6274-490b-bac6-90102f963749">

**Reply by mashb1t** [reply]
@ILikeToasters @xhoxye @Irmagon while waiting for the 4step CFG LoRA I've tested the 8step CFG one. You can find a preset draft here:
[hyper-sd8cfg.json](https://github.com/lllyasviel/Fooocus/files/15209936/hyper-sd8cfg.json)

This preset is not intended to be implemented as default, but just for evaluation.
The image is generated with quite a high amount of artifacts when using LoRA weight > 0.4, which is why 0.3 is used here. It's less about the actual CFG. Negative prompts do work, but, while being really good, i wouldn't necessarily say that it replaces normal SDXL 30+ step rendering as ByteDance claims on their HF page.

> we strongly recommend making the 8-step CFGLora a standard configuration for all SDXL and SD15 models

Happy to hear your thoughts on this one.


ps this is basically the same as just adding a new performance with 8 steps. Code see commit https://github.com/lllyasviel/Fooocus/pull/2812/commits/8ef1a2119c79de18c0e296088b2f2c1d6b6d36da.
Performance overview with examples => https://gallery.mashb1t.de/gallery/5aO0qQP1NQjp8r1SoVJKUIWc

**Reply by chouhai2018** [reply]
太实用了,我调好了,1秒1张1024x1024高清
![2024-05-05_19-46-18_5725](https://github.com/lllyasviel/Fooocus/assets/39079284/bbf41040-6da7-4f38-89dd-04e4fa86096d)

**Comment by mashb1t** [comment]
Hyper-SD 4 step LoRA is now part of release [2.4.0-rc1](https://github.com/lllyasviel/Fooocus/releases/tag/2.4.0-rc1).
Why not 8/12 step CFG LoRA? => improvement in image quality is not noticable (subjective) and i'm still waiting for the 4step CFG LoRA for best comparability.
Open ToDo's: see https://github.com/lllyasviel/Fooocus/discussions/2955#discussioncomment-9486937
